alacritty/src/config/monitor.rs

```diff
@@ -62,32 +64,68 @@ pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventLoopProxy<Event>) {
         parents.sort_unstable();
         parents.dedup();
 
-        let watcher = debouncer.watcher();
         // Watch all configuration file directories.
         for parent in &parents {
             if let Err(err) = watcher.watch(parent, RecursiveMode::NonRecursive) {
                 debug!("Unable to watch config directory {:?}: {}", parent, err);
             }
         }
 
+        // The current debouncing time.
+        let mut debouncing_time = None;
+
+        // The events accumulated during the debounce period.
+        let mut received_events = Vec::with_capacity(8);
+
         loop {
-            let events = match rx.recv() {
-                Ok(Ok(events)) => events,
+            // We use `recv_timeout` to debounce the events comming from the watcher and reduce
+            // the amount of config reloads.
+            let event = if let Some(debouncing_time) = debouncing_time.as_mut() {
+                let time_waited = Instant::now();
+                let event = rx.recv_timeout(*debouncing_time);
+                // Adjust the debouncing time, so we debounce at least each `DEBOUNCE_DELAY`.
+                *debouncing_time = debouncing_time.saturating_sub(time_waited.elapsed());
+                event
+            } else {
+                debouncing_time = Some(DEBOUNCE_DELAY);
+                rx.recv().map_err(Into::into)
+            };
+
+            match event {
+                Ok(Ok(event)) => {
+                    // Receiving any event should reset debouncing.
```

Comment:
It shouldn't and you're not actually resetting debouncing here? So the comment seems wrong?

Reply:
Yes, you're correct. I was resetting in the first impl, but decided that I should change that to not do so, to prevent infinite wait if you update config each `9ms` for example.